traffic_ops/traffic_ops_golang/invalidationjobs/invalidationjobs.go

```diff
@@ -228,10 +228,28 @@ func (job *InvalidationJob) Read(h http.Header, useIMS bool) ([]interface{}, err
 	if err != nil {
 		return nil, nil, fmt.Errorf("getting accessible tenants for user - %v", err), http.StatusInternalServerError, nil
 	}
+	cdn := ""
+	if cdnName, ok := job.APIInfo().Params["cdn"]; ok {
+		queryValues["cdn"] = cdnName
+		cdn = ` AND ds.cdn_id = (SELECT id FROM cdn WHERE name = :cdn) `
```

Comment:
fwiw, I think this actually could be a `dbhelpers.WhereInfo` if you did it like:
```go
"cdn": dbhelpers.WhereColumnInfo{Column: `(SELECT cdn.name FROM public.cdn WHERE cdn.id=ds.cdn_id)`},
```
but I haven't tested that, and what you have works fine.

Reply:
I tried that out at first, but the performance was atrocious. I think that subquery is run per row, whereas this version just runs it once to get the CDN ID.

Reply:
Oh. That's interesting. Wonder if the others done that way in that query are having similar problems.

Reply:
Possibly, but since `t3c` isn't using those filters I decided not to look into them.